app/controllers/declaration/CarrierDetailsController.scala

```diff
@@ -49,8 +51,8 @@ class CarrierDetailsController @Inject()(
   def displayPage(mode: Mode): Action[AnyContent] =
     (authenticate andThen journeyType(validTypes)) { implicit request =>
       request.cacheModel.parties.carrierDetails match {
-        case Some(data) => Ok(carrierDetailsPage(mode, navigationForm, form().fill(data)))
-        case _          => Ok(carrierDetailsPage(mode, navigationForm, form()))
+        case Some(data) => Ok(carrierDetailsPage(mode, navigationPage, form().fill(data)))
```

Comment:
Are you getting together with @MaciejRewera to get rid of this navigationPage nonsense?

Reply:
Yeah, I talked with him, but this PR will go first so I don't have the Navigator logic